pkg/tsdb/cloudwatch/credentials.go

```diff
@@ -149,16 +149,8 @@ func (e *CloudWatchExecutor) getDsInfo(region string) *DatasourceInfo {
 
 	authType := e.DataSource.JsonData.Get("authType").MustString()
 	assumeRoleArn := e.DataSource.JsonData.Get("assumeRoleArn").MustString()
-	accessKey := ""
-	secretKey := ""
-	for key, value := range e.DataSource.SecureJsonData.Decrypt() {
-		if key == "accessKey" {
-			accessKey = value
-		}
-		if key == "secretKey" {
-			secretKey = value
-		}
-	}
+	accessKey, _ := e.DataSource.DecryptedValue("accessKey")
```

Comment:
I think it would be more efficient to get the decrypted map just once.

Reply:
Yeah it's probably a good idea to avoid unnecessary locking :+1: